coral/src/app/features/approvals/acls/AclApprovals.test.tsx

```diff
@@ -0,0 +1,207 @@
+);
+
+describe("AclApprovals", () => {
+  beforeAll(() => {
+    mockIntersectionObserver();
+  });
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe("Skeleton", () => {
+    beforeAll(() => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      useQuerySpy.mockReturnValue({ data: [], isLoading: true });
+
+      customRender(<AclApprovals />, {
+        queryClient: true,
+        memoryRouter: true,
+      });
+    });
+
+    afterAll(() => {
+      cleanup();
+      useQuerySpy.mockRestore();
+    });
+
+    it("shows a skeleton table", () => {
+      const skeleton = screen.getByTestId("skeleton-table");
+
+      expect(skeleton).toBeVisible();
+    });
+  });
+
+  describe("DataTable", () => {
+    beforeAll(async () => {
+      mockGetAclRequestsForApprover.mockResolvedValue(
+        mockGetAclRequestsForApproverResponse
+      );
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      customRender(<AclApprovals />, {
+        queryClient: true,
+        memoryRouter: true,
+      });
+
+      await waitForElementToBeRemoved(screen.getByTestId("skeleton-table"));
+    });
+
+    afterAll(() => {
+      cleanup();
+    });
+
+    it("shows one header row and two data rows", async () => {
+      const rows = screen.getAllByRole("row");
+
+      expect(rows).toHaveLength(3);
+      expect(rows[0]).toHaveTextContent("Principals/Usernames");
+      expect(rows[1]).toHaveTextContent("mbasani");
+      expect(rows[2]).toHaveTextContent("User:*");
+    });
+
+    it("renders (prefixed) in Topic cell when appropriate", async () => {
+      const prefixedCells = screen.getAllByText("(prefixed)");
+      const notPrefixedCells = screen.getAllByText("aivtopic1");
+
+      expect(prefixedCells).toHaveLength(1);
+      expect(notPrefixedCells).toHaveLength(1);
+    });
+
+    it("renders all values for cells who can have multiple values", async () => {
+      const cells = screen.getAllByRole("cell");
+
+      expect(
+        cells.filter((cell) => {
+          return cell.textContent === "mbasanimaulbach";
+        })
+      ).toHaveLength(1);
+      expect(
+        cells.filter((cell) => cell.textContent === "3.3.3.323.3.3.33")
+      ).toHaveLength(1);
+    });
+  });
+
+  describe("Pagination", () => {
+    beforeAll(async () => {
+      mockGetAclRequestsForApprover.mockResolvedValue(
+        mockGetAclRequestsForApproverResponse
+      );
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      customRender(<AclApprovals />, {
+        queryClient: true,
+        memoryRouter: true,
+      });
+
+      await waitForElementToBeRemoved(screen.getByTestId("skeleton-table"));
+    });
+
+    afterAll(() => {
+      cleanup();
+    });
+
+    it("render a Pagination component", async () => {
+      const pagination = screen.getByRole("navigation");
+
+      expect(pagination).toBeInTheDocument();
```

Comment:
minor nitpick:
`toBeInTheDocument` is best used in combinations with `.not` or in rare cases where an element should be in the DOM but is not visible OR discoverable by assistive technology. This _could_ be a green test even if nobody can see that element. Also, `toBeEnabled` does indirectly confirm visibility for an element, otherwise it could not be enabled, so you could just remove that line.

Reply:
Yes, I don't know why I used this method, I guess I was feeling adventurous haha! Thank you for the explanation ^^

Reply:
[bb0ac03](https://github.com/aiven/klaw/pull/581/commits/bb0ac03fe5cf838801926500cb5a0fb10433ac26)